var/spack/repos/builtin/packages/py-pennylane-lightning/package.py

```diff
@@ -7,20 +7,118 @@
+    version(
+        "0.28.0",
+        sha256="f5849c2affb5fb57aca20feb40ca829d171b07db2304fde0a37c2332c5b09e18",
+        deprecated=True,
+    )  # on Spack v0.19.0
+
+    variant("blas", default=True, description="Build with BLAS support")
+    variant(
+        "dispatcher",
+        default=True,
+        description="Build with AVX2/AVX512 gate automatic dispatching support",
+    )
+    variant("kokkos", default=True, description="Build with Kokkos support")
+    variant("openmp", default=True, description="Build with OpenMP support")
+
+    variant("native", default=False, description="Build natively for given hardware")
+    variant("verbose", default=False, description="Build with full verbosity")
+
+    variant("cpptests", default=False, description="Build CPP tests")
+    variant("cppbenchmarks", default=False, description="Build CPP benchmark examples")
+
+    variant(
+        "build_type",
+        default="Release",
+        description="CMake build type",
+        values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
+    )
+
+    extends("python")
+
+    # hard dependencies
+    depends_on("cmake@3.21:3.24,3.25.2:", type="build")
+    depends_on("ninja", type=("run", "build"))
+
+    # variant defined dependencies
+    depends_on("blas", when="+blas")
+    depends_on("kokkos@3.7.00", when="+kokkos")
+    depends_on("kokkos-kernels@3.7.00", when="+kokkos")
+    depends_on("llvm-openmp", when="+openmp %apple-clang")
 
     depends_on("python@3.8:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("cmake@3.16:", type="build")
-
-    depends_on("py-ninja", type=("build", "run"))
     depends_on("py-numpy", type=("build", "run"))
-    # depends_on("py-pennylane@0.19:", type=("build", "run"))  # circular dependency
+    depends_on("py-pybind11", type=("build"))
+    depends_on("py-pip", type="build")
+    depends_on("py-wheel", type="build")
+    # depends_on("py-pennylane@0.28:", type=("build", "run"))  # circular dependency
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    build_directory = "build"
+
+    def cmake_args(self):
+        """
+        Here we specify all variant options that can be dynamicaly specified at build time
+        """
+        args = [
+            self.define_from_variant("CMAKE_BUILD_TYPE", "build_type"),
+            self.define_from_variant("ENABLE_OPENMP", "openmp"),
+            self.define_from_variant("ENABLE_NATIVE", "native"),
+            self.define_from_variant("ENABLE_BLAS", "blas"),
+            self.define_from_variant("CMAKE_VERBOSE_MAKEFILE:BOOL", "verbose"),
+            self.define_from_variant("BUILD_TESTS", "cpptests"),
+            self.define_from_variant("BUILD_BENCHMARKS", "cppbenchmarks"),
+            self.define_from_variant("ENABLE_GATE_DISPATCHER", "dispatcher"),
+        ]
+
+        if self.spec.variants["kokkos"].value:
```

Comment:
```suggestion
        if "+kokkos" in self.spec:
```
Simpler way of doing this

Reply:
Done.